analysis/dataset_definition_dm017.py

```diff
@@ -7,8 +8,10 @@
     get_dm_reg_r2,
 )
 
-# Define index date
-index_date = date(2022, 3, 31)
+parser = ArgumentParser()
+parser.add_argument("--index-date", type=str)
+args = parser.parse_args()
+index_date = datetime.strptime(args.index_date, '%Y-%m-%d').date()
```

Comment:
You can do the same thing with the `date.fromisoformat` function e.g:
```py
index_date = datetime.date.fromisoformat(args.index_date)
```

(This is assuming you have `import datetime` rather than `from datetime import datetime`. The fact that Python uses `datetime` as both the module and type name will never cease to be annoying.)